tests/unit/Rules/ContainsTest.php

```diff
@@ -19,8 +19,14 @@
  * @group  rule
  * @covers \Respect\Validation\Rules\Contains
  * @covers \Respect\Validation\Exceptions\ContainsException
+ *
+ * @author Alexandre Gomes Gaigalas <alexandre@gaigalas.net>
+ * @author Gabriel Caruso <carusogabriel34@gmail.com>
+ * @author Henrique Moody <henriquemoody@gmail.com>
+ * @author Nawarian <nickolas@phpsp.org.br>
+ * @author William Espindola <oi@williamespindola.com.br>
  */
-class ContainsTest extends TestCase
+final class ContainsTest extends TestCase
```

Comment:
Here we're doing usage of `RuleTestCase`. Could you do that change? [Example](https://github.com/Respect/Validation/pull/970/files).

Reply:
All of the test cases from this class could be written through the providers when inheriting `RuleTestCase`.

Reply:
Not in this case because the [Contains rule](https://github.com/Respect/Validation/blob/1.1/library/Rules/Contains.php#L19) has params on constructor.

Reply:
That's no problem, check out the [EqualsTest](https://github.com/Respect/Validation/blob/master/tests/unit/Rules/EqualsTest.php)

Reply:
nice!